cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java

```diff
@@ -1755,19 +1755,33 @@ public List<Condition> getWarningConditions() {
     }
 
     public DeploymentStrategy getDeploymentStrategy() {
-        if (templateDeploymentStrategy == io.strimzi.api.kafka.model.template.DeploymentStrategy.ROLLING_UPDATE) {
-            return new DeploymentStrategyBuilder()
-                    .withType("RollingUpdate")
-                    .withRollingUpdate(new RollingUpdateDeploymentBuilder()
-                            .withMaxSurge(new IntOrString(1))
-                            .withMaxUnavailable(new IntOrString(0))
-                            .build())
-                    .build();
-        } else {
-            return new DeploymentStrategyBuilder()
-                    .withType("Recreate")
-                    .build();
+        if (templateDeploymentStrategy == null) {
+            return buildRollingUpdateDeploymentStrategy();
+        }
+        switch (templateDeploymentStrategy) {
+            case ROLLING_UPDATE:
+                return buildRollingUpdateDeploymentStrategy();
+            case RECREATE:
+                return buildRecreateDeploymentStrategy();
+            default:
+                return buildRollingUpdateDeploymentStrategy();
         }
```

Comment:
But the old code defaulted to Recreate -> so it changes the behavior for anyone using this method.

Reply:
The method needs to have a `return` statement in every execution path. But when you write the code like this:

```
if
  ...
  return
else if
  ...
endif

...
return
```

It will pass the validation despite the possibly missing `return` in the `elseif` branch. If you write the code like this:

```
if
  ...
  return
else if
  ...
else
  ...
  return
endif
```

it will fail because there is no return.

This might seemingly not apply to your code. But next week someone might come, add the `elseif` to the method anf forget the return and it will _compile_ all fine. So using the full `if-else` construct will help to eliminate possible future bugs. (learned from experience and past bugs)

Reply:
Thanks for the explanation. Will add it.